ray-operator/controllers/ray/common/pod.go

```diff
@@ -793,12 +775,48 @@ func generateRayStartCommand(ctx context.Context, nodeType rayv1.RayNodeType, ra
 	return rayStartCmd
 }
 
-func addCustomAcceleratorToResourcesIfNotExists(rayStartParams map[string]string, resourceName string, resourceCount int64) error {
-	resourcesMap, err := getResourcesMap(rayStartParams)
-	if err != nil {
-		return err
+func addWellKnownAcceleratorResources(log logr.Logger, rayStartParams map[string]string, resourceLimits corev1.ResourceList) {
+	resourcesMap, _ := getResourcesMap(rayStartParams)
+
+	// Flag to track if any custom accelerator resource are present/added in rayStartParams resources.
+	isCustomAcceleratorResourceAdded := isCustomAcceleratorPresentInResources(resourcesMap)
+
+	for resourceKey, resourceValue := range resourceLimits {
+		resourceKeyString := string(resourceKey)
+
+		// Scan for resource keys ending with "gpu" like "nvidia.com/gpu"
+		if _, ok := rayStartParams["num-gpus"]; !ok {
+			if strings.HasSuffix(resourceKeyString, "gpu") && !resourceValue.IsZero() {
+				rayStartParams["num-gpus"] = strconv.FormatInt(resourceValue.Value(), 10)
+			}
+		}
+
+		// Add the first encountered custom accelerator resource from the resource limits to the rayStartParams if not already present
+		if resourcesMap != nil && !isCustomAcceleratorResourceAdded {
+			if rayResourceName, ok := customAcceleratorToRayResourceMap[resourceKeyString]; ok && !resourceValue.IsZero() {
+				if err := addCustomAcceleratorToResourcesIfNotExists(rayStartParams, resourcesMap, rayResourceName, resourceValue.Value()); err != nil {
```

Comment:
Can you avoid updating `rayStartParams` and `resourcesMap` in `addCustomAcceleratorToResourcesIfNotExists`, especially since this loop also checks `resourcesMap`? Maybe `addCustomAcceleratorToResourcesIfNotExists` is not necessary.

Reply:
Removed the function entirely and added the code in the same function